### DISCUSSION TITLE
My collection has been reset

### DISCUSSION THREAD

**Discussion by HelloYeew** [ppy/osu#13620] [Q&A] [answered]
I always play osu! on osu!lazer with my MacBook but today I open osu! like normally I open it but all of my collections are all gone. I checked on settings, beatmaps, scores is all fine and I found this issue on https://github.com/ppy/osu/discussions/13350 but smoogipoo fix it by send a new `collection.db` file and I check my `collection.db` on hex viewer and a count is show as zero like in https://github.com/ppy/osu/discussions/13350.

![CleanShot 2564-06-23 at 00 45 40](https://user-images.githubusercontent.com/68165621/122974352-66fb7500-d3bc-11eb-9c42-a17cb8ab8fef.png)

Every problem is like in https://github.com/ppy/osu/discussions/13350. How can I fix or recovery my collection back? Thx!

Here is my collection file.
[collection.db.zip](https://github.com/ppy/osu/files/6696380/collection.db.zip)

**Comment by peppy** [comment]
Which version did this occur on?

**Reply by HelloYeew** [reply]
Latest version (2021.619.1)

**Comment by smoogipoo** [comment, marked as answer]
You've actually experienced some data loss there. I've tried my best to recover it: [collection.db.zip](https://github.com/ppy/osu/files/6698875/collection.db.zip)

Note:
- The first beatmap hash is broken and I've excluded it (was a part of the `Bang Dream!` collection).
  - If you provide the `client.db` file, I may be able to restore it.
- There were 4 beatmaps that were outside of any collection at the end of the file. This could indicate a missing collection/incomplete file. I've grouped these 4 beatmaps into the `Warmup` collection.

It's interesting that the very first collection is as broken as it is... I think we'll have to save to a temporary location and do an in-place move rather than overwriting existing contents as an additional safeguard.

**Reply by HelloYeew** [reply]
It's back! Thanks for your help! This is enough.
